### PR TITLE
fix when cmake version<3.13, gRPC_INSTALL will be forced to FALSE bec…

### PR DESCRIPTION
**Pull request by walkerdu**
fix the BUILDING.md example code,  when cmake version<3.13, gRPC_INSTALL will be forced to FALSE because gRPC_ABSL_PROVIDER is "module"

@donnadionne

### CODE REVIEW

File: BUILDING.md [hunk 1]
```diff
@@ -217,7 +217,8 @@ $ cmake ../.. -DgRPC_INSTALL=ON                \
               -DgRPC_CARES_PROVIDER=package    \
               -DgRPC_PROTOBUF_PROVIDER=package \
               -DgRPC_SSL_PROVIDER=package      \
-              -DgRPC_ZLIB_PROVIDER=package
+              -DgRPC_ZLIB_PROVIDER=package     \
+              -DgRPC_RE2_PROVIDER=package
```
Comment (veblush) [inline BUILDING.md:221]: Can you move RE2 up to right below `DgRPC_PROTOBUF_PROVIDER` so that these are alphabetically ordered?

**Comment by veblush** [comment]
Thank you for the PR!

File: BUILDING.md (hunk 1 above)
Comment (walkerdu) [reply BUILDING.md]: Done, commit-id:   `2b73fcc`

Comment (veblush) [reply BUILDING.md]: Thanks!

**Review by veblush — APPROVED** [review]

**Comment by veblush** [comment]
@donnadionne, can you review this for another approval?

**Review by donnadionne — APPROVED** [review]

**Pull request merged**